core/src/main/java/org/apache/carbondata/core/scan/collector/impl/DictionaryBasedResultCollector.java

```diff
@@ -27,6 +28,7 @@
 import org.apache.carbondata.core.keygenerator.directdictionary.DirectDictionaryKeyGeneratorFactory;
 import org.apache.carbondata.core.metadata.datatype.DataTypes;
 import org.apache.carbondata.core.metadata.encoder.Encoding;
+import org.apache.carbondata.core.metadata.schema.table.column.CarbonDimension;
```

Comment:
To handle dictionary complex type, try to place this code in ColumnPageWrapper
```
@Override public int fillSurrogateKey(int rowId, int chunkIndex, int[] outputSurrogateKey,
      KeyStructureInfo restructuringInfo) {
    byte[] rowBytes = columnPage.getBytes(rowId);

    outputSurrogateKey[chunkIndex] = CarbonUtil.getSurrogateInternal(rowBytes, 0, rowBytes.length);
    return chunkIndex + 1;
  }
```

Reply:
Dictionary will be handled with Adaptive encoding PR